UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Fulfilment/Services/ExchangeSetApiService.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using UKHO.PeriodicOutputService.Common.Helpers;
+using UKHO.PeriodicOutputService.Fulfilment.Configuration;
+using UKHO.PeriodicOutputService.Fulfilment.Logging;
+using UKHO.PeriodicOutputService.Fulfilment.Models;
+
+namespace UKHO.PeriodicOutputService.Fulfilment.Services
+{
+    public class ExchangeSetApiService : IExchangeSetApiService
+    {
+        private readonly IOptions<ExchangeSetApiConfiguration> _exchangeSetApiConfiguration;
+        private readonly IExchangeSetApiClient _exchangeSetApiClient;
+        private readonly IAuthTokenProvider _authTokenProvider;
+        private readonly ILogger<ExchangeSetApiService> _logger;
+
+        public ExchangeSetApiService(ILogger<ExchangeSetApiService> logger,
+                                     IOptions<ExchangeSetApiConfiguration> exchangeSetApiConfiguration,
+                                     IExchangeSetApiClient exchangeSetApiClient, IAuthTokenProvider authTokenProvider)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _exchangeSetApiConfiguration = exchangeSetApiConfiguration ?? throw new ArgumentNullException(nameof(exchangeSetApiConfiguration));
+            _exchangeSetApiClient = exchangeSetApiClient ?? throw new ArgumentNullException(nameof(exchangeSetApiClient));
+            _authTokenProvider = authTokenProvider ?? throw new ArgumentNullException(nameof(authTokenProvider));
+        }
+
+        public async Task<ExchangeSetGetBatchResponse> GetProductIdentifiersData(List<string> productIdentifiers)
+        {
+            _logger.LogInformation(EventIds.AccessTokenForESSEndpointStarted.ToEventId(), "Getting access token to call ESS endpoint started");
+
+            string accessToken = await _authTokenProvider.GetManagedIdentityAuthAsync(_exchangeSetApiConfiguration.Value.EssClientId);
+
+            _logger.LogInformation(EventIds.AccessTokenForESSEndpointCompleted.ToEventId(), "Getting access token  to call ESS endpoint completed");
+
+            ExchangeSetGetBatchResponse exchangeSetGetBatchResponse = new();
+
+            _logger.LogInformation(EventIds.ExchangeSetRequestStarted.ToEventId(), "Request to get the exchange set details started");
+
+            HttpResponseMessage httpResponse = await _exchangeSetApiClient.GetProductIdentifiersDataAsync(_exchangeSetApiConfiguration.Value.BaseUrl, productIdentifiers, accessToken);
+
+            if (httpResponse.IsSuccessStatusCode)
+            {
+                string bodyJson = await httpResponse.Content.ReadAsStringAsync();
+                exchangeSetGetBatchResponse = JsonConvert.DeserializeObject<ExchangeSetGetBatchResponse>(bodyJson);
+            }
+            else
+            {
+                _logger.LogInformation(EventIds.ExceptionInExchangeSetRequest.ToEventId(), "Failed getting exchange set details");
```

Comment:
If the http response is not successful, should the logger output an error or warning rather than an information level log?

Reply:
Resolved. Changed to Error log.